### PR TITLE
Add RFC for pre-allocated tensors.

### PR DESCRIPTION
**Pull request by jenselofsson**
Add a RFC-document for the pre-allocated tensors feature.

### CODE REVIEW

File: tensorflow/lite/micro/docs/preallocated_tensors.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+The tensor arena is allocated by MicroAllocator in TensorFlow Lite Micro, and the model buffer (represented by a .tflite-file) is allocated by the application using TensorFlow Lite Micro. An illustration of this can be seen in the image below.
+
+![Image of two blocks](images/preallocated_tensors_bg_1.png)
+
+Is some use cases it could be advantageous to place some of the EvalTensors outside of the tensor arena, for example: 
+* When sensor output data is stored in its own defined buffer, outside the tensor arena, and therefore needs to be copied into the tensor arena before inference.
+* When the tensor is to be consumed from a memory location outside the tensor arena, e.g. a separate memory bank DSP.
```
Comment (jenselofsson) [inline tensorflow/lite/micro/docs/preallocated_tensors.md:10]: Add a line break after this line.

**Review by advaitjain — CHANGES_REQUESTED** [review]
looks good to me.

It might be worth a sub-folder in images called preallocated_tensors so that there is a bit of a directory structure there.

And could you also link to this from the top-level README (just add to the bottom of the list for now, and we'll reorganize later):
https://github.com/tensorflow/tensorflow/blob/78fdd635abb65422c5a4b72750e9fafccf8bee61/tensorflow/lite/micro/README.md#L65-L69

**Comment by gbaned** [comment]
@jenselofsson  Can you please check @advaitjain's comments and keep us posted ? Thanks!

**Comment by jenselofsson** [comment]
> looks good to me.
> 
> It might be worth a sub-folder in images called preallocated_tensors so that there is a bit of a directory structure there.
> 

Good idea!

> And could you also link to this from the top-level README (just add to the bottom of the list for now, and we'll reorganize later):
> 
> https://github.com/tensorflow/tensorflow/blob/78fdd635abb65422c5a4b72750e9fafccf8bee61/tensorflow/lite/micro/README.md#L65-L69

Sure!

**Review by advaitjain — APPROVED** [review]

**Comment by advaitjain** [comment]
copybara failed, triggering it again.

**Pull request merged**